README.md

```diff
@@ -6,6 +6,158 @@ It can make sure, for example, that changes are attributed properly, have a vali
+### Colons: They are awesome
+#### Probably not used all that often
+```
+
+❌ Only letters (a-z), numbers, underscores and colons are allowed in titles
+
+```markdown
+#### Emoji is not allowed 😢
+#### Hyphens are not allowed thus sugar-free is not valid
+#### Multiple sentences. Are not allowed
+#### Commas, neither
+#### No accented characters sorry Beyoncé
+```
+
+❌ Titles must start with a leading space
+
+```markdown
+#This is not valid
+```
+
+✅ You can place your titles inside a single parens
+
+```markdown
+# (This is allowed)
+## [This is allowed]
+### {This is also allowed}
+```
+
+❌ Parens in a title must match, and you can only have one set
+
+```markdown
+# (Not valid
+## {Not valid)
+### Not valid ()
+#### [] Not valid
+#### ((Not valid))
+```
+
+✅ You can have a [semantic-versioning](https://semver.org/) compatible version number
+
+```markdown
+# 1.0.0
+## 1.0.0-alpha
+### (3.2.1-beta+reallybuggy.1)
+```
+
+❌ Version numbers which do not follow [semantic-versioning](https://semver.org/) are not allowed
+
+```markdown
+# 01.0.0.notvalid
+```
+
+❌ You cannot combine version numbers with other text
+
+```markdown
+# Not valid 1.0.0
+# Not valid (1.0.0)
+```
+
+✅ You can include an [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601)-formatted date along with a version, inside or outside parens
+
+```markdown
+# 1.0.0 2019
+## 1.0.0 (2019-12-31)
+### (1.0.0) (2019/12/31)
+#### (1.0.0) 20191231
+```
+
+❌ No other date or time formats are allowed
+
+```markdown
+# 1.0.0 (2019-12)
+# 1.0.0 (2019-W1)
+# 1.0.0 (2019-04-28T09:21:58Z)
+# 1.0.0 (31/12/2019)
+```
+
+❌ Dates cannot be combined with normal titles
+
+```markdown
+# Not valid (2019/12/31)
+```
+
+### Changelog entries
+
+✅ Under each title, you can have changelog entries in the format: `* {optional link to GitHub pull request that must be followed by a colon} {a single sentence of text description} - {author github link}.`:
+
+```markdown
+* Entry without a PR link - [@ivoanjo](https://github.com/ivoanjo).
+* [#1](https://github.com/dblock/danger-changelog/pull/1): Entry with a PR link - [@dblock](https://github.com/dblock).
+* Yes, this is a weird entry, but it's still valid! Also Beyoncé is welcome here, as is 松本 行弘 and our nice emojis 👍 - [@ivoanjo](https://github.com/ivoanjo).
+```
+
+❌ Entry descriptions need to start with an uppercase A-Z letter
+
+```markdown
+* not valid - [@ivoanjo](https://github.com/ivoanjo).
+* Ñot valid - [@ivoanjo](https://github.com/ivoanjo).
+* ❌ - [@ivoanjo](https://github.com/ivoanjo).
+* [#1](https://github.com/dblock/danger-changelog/pull/1): not valid - [@dblock](https://github.com/dblock).
+```
+
+❌ GitHub pull request links need to be followed with a colon
+
+```markdown
+* [#1](https://github.com/dblock/danger-changelog/pull/1) Not valid - [@dblock](https://github.com/dblock).
+```
+
+❌ Only links to GitHub are allowed, and only using `https`
+
+```markdown
+* No GitLab - [@ivoanjo](https://gitlab.com/ivoanjo).
```

Comment:
This looks like a bug too, we should support gitlab!

Reply:
I'll report it 👍